src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -636,7 +635,7 @@ else if (customImportOrderRules.contains(SAME_PACKAGE_RULE_GROUP)) {
                 }
                 if (SPECIAL_IMPORTS_RULE_GROUP.equals(group)) {
                     bestMatch = findBetterPatternMatch(importPath,
-                            SPECIAL_IMPORTS_RULE_GROUP, specialImportsRegExp, bestMatch);
+                            group, specialImportsRegExp, bestMatch);
```

Comment:
Can you explain why this change is needed for pitest?
report didn't mention anything about this, only the if condition. 
If it is not needed, please restore the original.

Reply:
> report didn't mention anything about this, only the if condition.

Right, when we negate if condition and `group != SPECIAL_IMPORTS_RULE_GROUP`, `findBetterPatternMatch` method works right with `SPECIAL_IMPORTS_RULE_GROUP` paraｍeter which means if condition meaningless.
 But, obviously, the if condition can't be omitted. So I do this change.

Reply:
> the if condition can't be omitted
> when we negate ... method works right with SPECIAL_IMPORTS_RULE_GROUP paraｍeter

There is no other way around this, like with a custom `specialImportsRegExp`?
`STANDARD_JAVA_PACKAGE_RULE_GROUP` doesn't have this issue and it has very similar code.

Reply:
> There is no other way around this

I have another way(add some tests) to fix it, but I think this code logic is better.

Reply:
@romani What is your opinion?

Reply:
I am fine with update.